test/units/resttests.ts

```diff
@@ -64,6 +64,28 @@ describe('Rest Tests', function () {
         assert(restRes.result && restRes.result.url === 'http://microsoft.com');
     });
 
+    it('gets a resource and correctly revives its Date property', async() => {
+        const dateObject: Date = new Date();
+        nock('http://microsoft.com')
+            .get('/date')
+            .reply(200, {
+                json: {dateProperty: dateObject.toDateString()}
+            });
+        const restRes: restm.IRestResponse<HttpData> = await _rest.get<HttpData>('http://microsoft.com/date', {reviveDates: true});
+        assert(restRes.result && restRes.result.json.dateProperty.getTime != undefined);
```

Comment:
assert.equals. Also, can we be more specific with the test? let's use a hard coded datetime and check all the properties of it.

Reply:
Fixed!